tests/epyccel/test_builtins.py

```diff
@@ -369,3 +369,72 @@ def f():
     epyc_f = epyccel(f, language=language)
 
     assert epyc_f() == f()
+
+@pytest.fixture( params=[
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="Set declaration in Fortran is not yet implemented. related issue #1658"),
+            pytest.mark.fortran])
+    ],
+    scope = "module"
+)
+def stc_language(request):
+    return request.param
+
+def test_len_list_int(stc_language):
+    def f():
+        a = [1, 2, 3]
+        return len(a)
+
+    epyc_f = epyccel(f, language=stc_language)
+
+    assert epyc_f() == f()
+
+def test_len_list_float(stc_language):
+    def f():
+        a = [1.4, 2.6, 3.5]
+        b = len(a)
+        return b
+
+    epyc_f = epyccel(f, language=stc_language)
+
+    assert epyc_f() == f()
+
+def test_len_list_complex(stc_language):
+    def f():
+        a = [1j, 2 + 1j, 3 + 1j]
+        b = len(a)
+        return b
+
+    epyc_f = epyccel(f, language=stc_language)
+
+    assert epyc_f() == f()
+
+def test_len_set_int(stc_language):
+    def f():
+        a = {1, 2, 3}
+        return len(a)
+
+    epyc_f = epyccel(f, language=stc_language)
+
+    assert epyc_f() == f()
+
+def test_len_set_float(stc_language):
+    def f():
+        a = {1.4, 2.6, 3.5}
+        b = len(a)
+        return b
+
+    epyc_f = epyccel(f, language=stc_language)
+
+    assert epyc_f() == f()
+
+def test_len_set_complex(stc_language):
+    def f():
+        a = {1j, 2 + 1j, 3 + 1j}
+        b = len(a)
+        return b
+
+    epyc_f = epyccel(f, language=stc_language)
+
+    assert epyc_f() == f()
```

Comment:
should we add more tests? especially for nested lists?

Reply:
Im really not sure about the handling of nested lists, if they weren't supported by default in the previous PR, we can open an issue for them asap

Reply:
Nested lists should be supported. I believe they are working for Python. Nested sets won't be supported (at least for now)

Reply:
I am working on nested lists. They are working for Python but there are some problems in C so it is probably not a good idea to add lots of extra tests for this immediately. In any case the way the `len` function works does not change depending on the element type so I don't think that lots of nested list tests would necessarily add much